src/components/Query.ts

```diff
@@ -21,36 +23,43 @@ export class ViewEngine extends Engine {
     this.tableView = new TableView(context)
   }
 
-  public async TableView (connection?: InfluxDBConnection, query?: string) {
-    const { activeTextEditor } = window
+  private showNoConnectionWarning () {
+    window.showWarningMessage('No influxDB Server selected')
+  }
 
-    if (!query && !activeTextEditor) {
-      return window.showWarningMessage('No Flux file selected')
-    }
+  private showNoFluxWarning () {
+    return window.showWarningMessage('No Flux file selected')
+  }
 
-    connection = connection || Status.Current
+  private get query () {
+    const { activeTextEditor } = window
 
-    if (!connection) {
-      window.showWarningMessage('No influxDB Server selected')
+    if (!activeTextEditor) {
       return
     }
 
-    if (!query && activeTextEditor) {
-      if (activeTextEditor.selection.isEmpty) {
-        query = activeTextEditor.document.getText()
-      } else {
-        query = activeTextEditor.document.getText(activeTextEditor.selection)
-      }
+    if (activeTextEditor.selection.isEmpty) {
+      return activeTextEditor.document.getText()
     }
 
-    query = query || ''
+    return activeTextEditor.document.getText(activeTextEditor.selection)
+  }
+
+  public async showTable () {
+    if (!this.query) {
+      return this.showNoFluxWarning()
+    }
+
+    if (!Status.Current) {
+      window.showWarningMessage('No influxDB Server selected')
+      return
+    }
 
-    logger.log(`Running Query: '${query}'`)
-    logger.show()
+    logger.log(`Running Query: '${this.query}'`)
 
     try {
-      const results = await APIRequest.query(connection, query)
-      return this.tableView.show(results.tables, connection.name)
+      const results = await APIRequest.query(Status.Current, this.query)
+      return this.tableView.show(results.tables, Status.Current?.name)
```

Comment:
this is interesting, I didn't know you can do `?` for a method access directly

Reply:
new typescript feature